### PR TITLE
Try optimize traffic reports

### PR DESCRIPTION
**Pull request by vincentlepot**
# Description

This PR tries to optimize a little traffic_reports calls.
On my env, I go from 8s to 2.5s. But not sure it will be the same in production...

## Issue (optional)

Issue link:

## Pull Request type (optional)

This is an attempt for performance improvement

## How to test

Here are the following steps to test this pull request:

- Try to use traffic_reports, it should be (a little) quicker
- Nothing should be broken

## Team reviewers (optional)

@CanalTP/bot

### CODE REVIEW

File: chaos/resources.py [hunk 1]
```diff
@@ -964,7 +964,21 @@ def get(self, contributor, navitia):
         args = self.parsers['get'].parse_args()
         g.current_time = args['current_time']
         disruptions = models.Disruption.traffic_report_filter(contributor.id)
-        result = utils.get_traffic_report_objects(disruptions, self.navitia)
+
+        # Prepare line sections to get them all in once
+        pt_object_ids = []
+        for disruption in disruptions:
+            for impact in disruption.impacts:
```
Comment (datanel) [inline chaos/resources.py]: I think you can do
```python
for impact in disruption.impacts if impact.status == 'published':
```

File: tests/features/impacts-with-pt-objects.feature [hunk 2]
```diff
@@ -507,9 +507,9 @@ Feature: Manipulate impacts in a Disruption
         And the field "impact.objects.0.line_section.start_point.id" should be "stop_area:JDR:SA:PTVIN"
         And the field "impact.objects.0.line_section.end_point.id" should be "stop_area:JDR:SA:BERAU"
         And the field "impact.objects.0.line_section.routes.0.type" should be "route"
-        And the field "impact.objects.0.line_section.routes.0.id" should be "route:JDR:M14"
+        And the field "impact.objects.0.line_section.routes.0.id" should be "route:JDR:M1"
```
Comment (pthegner) [inline tests/features/impacts-with-pt-objects.feature]: You could possibly use : 
```
        And the field "impact.objects.0.line_section.routes" should have a size of 2
        And the field "impact.objects.0.line_section.routes" should contain all of "{"type": "route", "id": "route:JDR:M1"}"
        And the field "impact.objects.0.line_section.routes" should contain all of "{"type": "route", "id": "route:JDR:M14"}"
```

File: chaos/resources.py (hunk 1 above)
Comment (vincentlepot) [inline chaos/resources.py]: Not directly (it leads to a syntax error). But maybe with a filter..

**Comment by vincentlepot** [comment]
Code updated with @pthegner feedback (thanks for that :) )

File: chaos/resources.py (hunk 1 above)
Comment (datanel) [inline chaos/resources.py]: and
```python
for impact in (i for in disruption.impacts if i.status == 'published'):
```
?

Comment (vincentlepot) [inline chaos/resources.py]: It looks better. Not sure it speeds up anything, but if you prefer ;)

Comment (vincentlepot) [inline chaos/resources.py]: Done

**Comment by vincentlepot** [comment]
Code updated with @datanel feedback. Thanks!

**Comment by vincentlepot** [comment]
If it's ok, can someone put a green light and merge?

**Pull request merged**